autopilot/contractor.go

```diff
@@ -100,17 +180,20 @@ func (c *contractor) performContractMaintenance(cfg Config) error {
 	}
 
 	// update contract set
-	err = c.ap.updateDefaultContracts(active, toRenew, renewed, formed, toDelete)
+	err = c.ap.updateDefaultContracts(active, renewed, formed, toRenew, toIgnore, toDelete)
 	if err != nil {
 		return fmt.Errorf("failed to update default contracts, err: %v", err)
 	}
 
 	return nil
 }
 
-func (c *contractor) runContractChecks(cfg Config, contracts []bus.Contract) ([]bus.Contract, []types.FileContractID, error) {
-	// collect contracts to renew and to delete
-	toDelete := make([]types.FileContractID, 0, len(contracts))
+// runContractChecks performs a series of checks on the given contracts and
+// splits them over three buckets, contracts to renew, archive or delete
```

Comment:
Isn't archive == delete?

Reply:
I wasn't aware you already implemented deletion on db level. Archive was a bad name for contracts that should be "ignored" (read: excluded) from the contract set. So they aren't in the contract set but they aren't deleted either.